spec/services/finance/declaration_statement_attacher_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+RSpec.describe Finance::DeclarationStatementAttacher do
+  let(:declaration) { create(:ect_participant_declaration, cpd_lead_provider: cpd_lead_provider) }
+  let(:statement) { create(:ecf_statement, output_fee: true, deadline_date: 2.months.from_now) }
+  let(:cpd_lead_provider) { statement.cpd_lead_provider }
+
+  subject { described_class.new(participant_declaration: declaration) }
+
+  describe "#call" do
+    it "creates line item" do
+      expect {
+        subject.call
+      }.to change(Finance::StatementLineItem, :count).by(1)
```

Comment:
slightly more robust, checks it is attached to the correct statement:
```suggestion
      }.to change(statement.line_items, :count).by(1)
```